llvm/utils/repo/elf.cmake

```diff
@@ -0,0 +1,57 @@
+#*       _  __  *
+#*   ___| |/ _| *
+#*  / _ \ | |_  *
+#* |  __/ |  _| *
+#*  \___|_|_|   *
+#*              *
+#===- elf.cmake -----------------------------------------------------------===//
+# Copyright (c) 2017-2019 by Sony Interactive Entertainment, Inc.
+# All rights reserved.
+#
+# Developed by:
+#   Toolchain Team
+#   SN Systems, Ltd.
+#   www.snsystems.com
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the
+# "Software"), to deal with the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# - Redistributions of source code must retain the above copyright notice,
+#   this list of conditions and the following disclaimers.
+#
+# - Redistributions in binary form must reproduce the above copyright
+#   notice, this list of conditions and the following disclaimers in the
+#   documentation and/or other materials provided with the distribution.
+#
+# - Neither the names of SN Systems Ltd., Sony Interactive Entertainment,
+#   Inc. nor the names of its contributors may be used to endorse or
+#   promote products derived from this Software without specific prior
+#   written permission.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE CONTRIBUTORS OR COPYRIGHT HOLDERS BE LIABLE FOR
+# ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+# SOFTWARE OR THE USE OR OTHER DEALINGS WITH THE SOFTWARE.
+#===----------------------------------------------------------------------===//
+
+set (triple "x86_64-pc-linux-gnu")
+
+# The user must specify CMAKE_C_COMPILER by giving '-DCMAKE_C_COMPILER:FILEPATH=/path/to/c/compiler' on the command line
+SET (CMAKE_C_COMPILER   "clang"  CACHE FILEPATH "Compiler")
+# The user must specify CMAKE_CXX_COMPILER by using '-DCMAKE_CXX_COMPILER:FILEPATH=/path/to/cxx/compiler' on the command line
+SET (CMAKE_CXX_COMPILER "clang++"  CACHE FILEPATH "Compiler")
+
+SET (CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -O0 -fno-exceptions -fno-rtti" CACHE STRING "Default C Flags Debug")
+SET (CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -O0 -fno-exceptions -fno-rtti" CACHE STRING "Default CXX Flags Debug")
+SET (CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -ftime-trace -O3 -fno-exceptions -fno-rtti" CACHE STRING "Default C Flags Release")
+SET (CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -ftime-trace -O3 -fno-exceptions -fno-rtti" CACHE STRING "Default CXX Flags Release")
```

Comment:
I'm not sure about force-enabling `-ftime-trace` for all of the builds we do. Wouldn't it be more standard practise to enable at configure-time by passing `-D CMAKE_C_FLAGS_xxx=-ftime-trace` when you want to capture timing from a build?

Reply:
I agree. I'll remove it.